prow/entrypoint/run.go

```diff
@@ -70,6 +76,16 @@ var (
 	// errAborted is used as the command's error when the command
 	// is shut down by an external signal
 	errAborted = errors.New("process aborted")
+
+	amazonKeychain authn.Keychain = authn.NewKeychainFromHelper(ecr.NewECRHelper(ecr.WithLogOutput(ioutil.Discard)))
+	azureKeychain  authn.Keychain = authn.NewKeychainFromHelper(credhelper.NewACRCredentialsHelper())
+	keychain                      = authn.NewMultiKeychain(
+		amazonKeychain,
```

Comment:
The ordering matters here, so this will first check if AWS creds can be discovered in the environment, _then_ check if there's a `~/.config/docker.json` available that provides creds. Just checking if this preference ordering was intentional, and if not, I might suggest putting `authn.DefaultKeychain` first since I think that's what most users would expect.

Reply:
makes sense, updated